File: BE/src/main/java/com/haivn/common_api/NhatKyThucTap.java
package com.haivn.common_api;
import lombok.Getter;
import lombok.Setter;
import org.hibernate.annotations.DynamicUpdate;
import org.hibernate.annotations.Where;
import javax.persistence.*;
import java.sql.Timestamp;

@Entity
@Table(name = "nhat_ky_thuc_tap")
@Getter
@Setter
@DynamicUpdate
@Where(clause = "deleted=false")
public class NhatKyThucTap extends BaseEntity{
    @Column(name = "id_sv")
    private Long idSv;
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name="id_sv",referencedColumnName="id", nullable = false, insertable = false, updatable = false)
    private SinhVienThucTap sinhVienThucTap;
    @Column(name = "name")
    private String name;
    @Column(name = "statr_date")
    private Timestamp startDate;
    @Column(name = "end_date")
    private Timestamp endDate;
    @Column(name = "content")
    private String content;
    @Column(name = "ghi_chu")
    private String ghiChu;
    @Column(name = "ket_qua")
    private String ketQua;
    @Column(name = "file")
    private String file;
    @Column(name = "status")
    private Short status;
}

File: fe_web/lib/models/nhat-ky-thuc-tap.dart
// ignore_for_file: file_names

class NhatKyTT {
  int? id;
  int? idSv;
  String? modifiedDate;
  String? name;
  String? startDate;
  String? endDate;
  String? content;
  String? ghiChu;
  String? ketQua;
  String? file;
  int? status;
  NhatKyTT({
    this.id,
    this.idSv,
    this.modifiedDate,
    this.name,
    this.startDate,
    this.endDate,
    this.content,
    this.ghiChu,
    this.ketQua,
    this.file,
    this.status,
  });

  NhatKyTT.fromJson(Map<String, dynamic> json)
      : id = json['id'],
        name = json['name'],
        modifiedDate = json['modifiedDate'],
        idSv = json['idSv'],
        startDate = json['startDate'],
        endDate = json['endDate'],
        ghiChu = json['ghiChu'],
        content = json['content'],
        ketQua = json['ketQua'],
        file = json['file'],
        status = json['status'];

  Map<String, dynamic> toJson() => {
        'name': name,
        'idSv': idSv,
        'startDate': startDate,
        'endDate': endDate,
        'content': content,
        'ghiChu': ghiChu,
        'ketQua': ketQua,
        'file': file,
        'status': status
      };
}
